### PR TITLE
Fix: Wait for syncsClient.run( before sending response.

### PR DESCRIPTION
**Pull request by irohitb**
Still trying to comprehend the code flow but `syncsClient.run` happens to be an async function. 

```
let syncResult = syncsClient.run(syncId, params.frequency);
```

this would send unresolved promise?  `res.status(200).send({ sync_id: syncId, result: syncResult });` if we don't use async?

### CODE REVIEW

File: packages/worker/lib/services/data.service.ts [hunk 1]
```diff
@@ -6,31 +6,20 @@ import { NangoColumnDataTypes } from '../models/data.types.js';
 
 class DataService {
     async upsertRawFromList(objects: RawObject[], sync: Sync): Promise<void | number[]> {
+        const query = db.knex<RawObject>(`_nango_raw`).withSchema(db.schema()).where('sync_id', sync.id);
```
Comment (bastienbeurier) [inline packages/worker/lib/services/data.service.ts:9]: Much cleaner!

File: packages/worker/lib/services/data.service.ts [hunk 2]
```diff
@@ -134,35 +113,17 @@ class DataService {
     }
 
     async createSyncTableIfNeeded(syncId: number) {
-        return new Promise<void>((resolve, reject) => {
```
Comment (bastienbeurier) [inline packages/worker/lib/services/data.service.ts:137]: Same here, not sure why I created a Promise here. Thanks!

File: packages/worker/lib/services/data.service.ts [hunk 3]
```diff
@@ -134,35 +113,17 @@ class DataService {
...
+        if (!exists) {
+            logger.debug(`Table ${this.tableNameForSync(syncId)} doesn't exist, creating the new table for Sync ID: ${syncId}.`);
+            await db.knex.schema.withSchema(db.schema()).createTable(this.tableNameForSync(syncId), (t) => {
+                t.increments('_nango_id').primary();
+                t.integer('_nango_sync_id').references('id').inTable(`${db.schema()}._nango_syncs`);
+                t.dateTime('_nango_emitted_at').notNullable();
+                t.string('_nango_unique_key');
+            });
+        }
+        return;
```
Comment (bastienbeurier) [inline packages/worker/lib/services/data.service.ts:126]: Unnecessary return statement?

**Review by bastienbeurier — APPROVED** [review]

File: packages/worker/lib/services/data.service.ts (hunk 3 above)
Comment (irohitb) [reply packages/worker/lib/services/data.service.ts]: Removed return. 

Waiting for someone to re-review and merge

**Pull request merged**